### DISCUSSION TITLE
Location of pictures

### DISCUSSION THREAD

**Discussion by KosuzuMotoori** [codingjoe/django-pictures#79] [Q&A]
Is it possible to save the thumbnails in a different directory than the original images? All the original images in one directory and all their thumbnails of different sizes in a separate directory for example "cache"?

/media/pictures/album/photo_001.jpg
/media/cache/album/photo_001/1800w.jpeg
/media/cache/album/photo_001/900w.jpeg
/media/cache/album/photo_001/300w.jpeg

That way it would be easier to delete all dynamically created photos and generate new ones.

**Comment by codingjoe** [comment]
Hi @KosuzuMotoori,

Thank you for reaching. Interesting question… I think I need a little more information to understand your use case better. But I will give a bit more detailed information about some things, even if they don't fully answer your question yet.

The file structure right now is not changeable via a setting. However, you can overwrite the name or path property, with a custom implementation: https://github.com/codingjoe/django-pictures/blob/2eed4283acc808632fd36cacbe61f172b98429c2/pictures/models.py#L57-L66

Regarding deletion. You can delete all sizes at once via `delete_all`, https://github.com/codingjoe/django-pictures/blob/2eed4283acc808632fd36cacbe61f172b98429c2/pictures/models.py#L105-L110

Since larger file storage like S3 are key-value stores and operate via API, I am not really sure if there is much gained from a differently prefixed location.

All that aside, if you want to just delete orphaned files, consider using `django-cleanup`. A simple `pip install django-pictures[cleanup]`  will do the trick.

I hope some of those remarks already helped you!

Best,
Joe

**Comment by KosuzuMotoori** [comment]
Thank you for your reply. I don't use cloud services, and if the original files are separate from the dynamically generated ones, it makes backup a little easier. This is important if there are a lot of pictures, in my case tens of gigabytes. 4-6 versions of the same image in different sizes increase the size of the backup significantly.

At the moment I'm using django-imagekit. It stores thumbnails files in a separate cache directory and I think it's handy.

Do you have any plans to implement management commands to regenerate and delete all thumbnails?

**Reply by codingjoe** [reply]
> Thank you for your reply. I don't use cloud services, and if the original files are separate from the dynamically generated ones, it makes backup a little easier. This is important if there are a lot of pictures, in my case tens of gigabytes. 4-6 versions of the same image in different sizes increase the size of the backup significantly.
> 
> At the moment I'm using django-imagekit. It stores thumbnails files in a separate cache directory and I think it's handy.
> 
> Do you have any plans to implement management commands to regenerate and delete all thumbnails?

I think you have an interesting use case. This package is kind of build on the principle that storage is cheap. So, we pre-generate all pictures instead of creating them on demand and caching them. However, was considering adding such a feature, but with a focus on CDN support.

I am not really planing on adding any management commands yet. Mainly because we have migration support, I haven't actually seen a need for that. Of course, you are always welcome to contribute something.

Since you are going more of a caching route. That is something that I want to implement in the next version. I'd absolutely love some input here.

Cheers!
Joe
